packages/opencode/src/tool/websearch.ts

```diff
@@ -0,0 +1,60 @@
+import { tool } from "@opencode-ai/plugin/tool"
+import { join } from "path"
+
+async function getPerplexityKey() {
+  const authPath = join(process.env.HOME || ".", ".local/share/opencode/auth.json")
+  const file = Bun.file(authPath)
+  if (!(await file.exists())) return null
+  const authData = await file.json()
+  return authData.PERPLEXITY_API_KEY || null
+}
+
+export const websearch = tool({
+  description: "Search the web for current information using Perplexity AI",
+  args: {
+    query: tool.schema
+      .string()
+      .describe("The search query"),
+  },
+  async execute(args) {
+    const apiKey = await getPerplexityKey()
+    
+    if (!apiKey) {
+      return "Error: Perplexity API key not found. Please set PERPLEXITY_API_KEY in ~/.local/share/opencode/auth.json"
+    }
+
+    const response = await fetch("https://api.perplexity.ai/chat/completions", {
+      method: "POST",
+      headers: {
+        "Authorization": `Bearer ${apiKey}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        model: "llama-3.1-sonar-small-128k-online",
```

Comment:
It could be better to just use more future-proof models like [`sonar`](https://docs.perplexity.ai/api-reference/chat-completions-post#body-model) or something?